server/client.go

```diff
@@ -591,6 +640,17 @@ func (c *client) processSub(argo []byte) (err error) {
 		return nil
 	}
 
+	// Check permissions if applicable.
+	if c.perms != nil && c.perms.sub != nil {
```

Comment:
As of now, it looks like if `c.perms` is not `nil`, then `perms.sub` and `perms.pub` will be not `nil`..